### DISCUSSION TITLE
Return only messages instead of the full page

### DISCUSSION THREAD

**Discussion by savchenko** [pallets-eco/flask-security#1005] [Q&A]
I am using HTMX and would rather add what's currently served by `get_flashed_messages(with_categories=true)` instead of re-rendering the whole page.

Is there a clever way to _not_ return a whole body after POST? Login page would be a good example.

**Comment by jwag956** [comment]
The basic idea for single-page-applications is that they should use the JSON API rather than the stock form-based templates. There is documentation about this in the readthedocs for Flask-Security. While there is initial extra effort to do this - is allows the application to manage routing, errors, UX consistently.

**Reply by savchenko** [reply]
Appreciate that, but mine is not a SPA. Frontend sends POST request and expects HTML that can be swapped into DOM.

**Comment by jwag956** [comment]
Also note that flashing are almost always used when redirecting - normal form errors in inline.
